lib/elixir/pages/typespecs.md

```diff
@@ -134,7 +134,7 @@ Built-in type           | Defined as
 `function()`            | `fun()`
 `identifier()`          | `pid()` \| `port()` \| `reference()`
 `iodata()`              | `iolist()` \| `binary()`
-`iolist()`              | `maybe_improper_list(byte() \| binary() \| iolist(), binary() \| [])`
+`iolist()`              | `maybe_improper_list(byte() \| binary() \| iolist(), binary())`
```

Comment:
I think this is incorrect. Otherwise it would always be an improper list.

Reply:
I think the `maybe_improper_list` implies that the second element is a `| []`?

Reply:
@josevalim I see it that way.

I think it should either be  `maybe_improper_list(..., binary())` or `list(.., binary() | [])` but t:list/2 does not exist.

Reply:
That is not how OTP treats the type:

* [`iolist()`](https://github.com/erlang/otp/blob/f8481a4bac07bc224e59629fda7e887411b40bcf/erts/preloaded/src/erlang.erl#L94)
* [`charlist()`](https://github.com/erlang/otp/blob/40922798411c2d23ee8a99456f96d6637c62b762/lib/stdlib/src/unicode.erl#L41)
* [dialyzer implementation](https://github.com/erlang/otp/blob/008f44fefeb950bc4bfbe6a13922f2a8f5af49c9/lib/dialyzer/src/erl_types.erl#L2267-L2271)